options/options.go

```diff
@@ -630,14 +647,16 @@ func (interp *interpreter) interpretOptions(
 			}
 		}
 		mc.Option = uo
+		interp.enableLenience(true)
 		srcInfo, err := interp.interpretField(targetType, mc, msg, uo, 0, interp.pathBuffer)
+		interp.enableLenience(false)
 		if err != nil {
-			if interp.lenient {
-				remain = append(remain, uo)
-				continue
-			}
 			return nil, err
 		}
+		if interp.lenientErrReported {
+			remain = append(remain, uo)
+			continue
+		}
```

Comment:
When lenience is enabled, the above error is always nil, but we still need to know if any error was reported, so we know whether we need to keep this option in the uninterpreted set.